SimCalorimetry/EcalSimAlgos/src/EcalTimeMapDigitizer.cc

```diff
@@ -55,11 +53,17 @@ EcalTimeMapDigitizer::EcalTimeMapDigitizer(EcalSubdetector myDet) : m_subDet(myD
   assert(m_minBunch <= 0);
 
   m_vSam.reserve(size);
+  m_index.reserve(size);
 
   for (unsigned int i(0); i != size; ++i) {
     //       m_vSam.emplace_back(CaloGenericDetId( detId.det(), detId.subdetId(), i ) ,
     // 			  m_maxBunch-m_minBunch+1, abs(m_minBunch) );
-    m_vSam.emplace_back(TimeSamples(CaloGenericDetId(detId.det(), detId.subdetId(), i)));
+    if      (myDet == EcalBarrel) {
+      m_vSam.push_back(TimeSamples((DetId)(EBDetId::detIdFromDenseIndex(i))));
+    }
+    else if (myDet == EcalEndcap) {
```

Comment:
Can `myDet` be anything else than `EcalBarrel` or `EcalEndcap`? If not maybe just using `else` is enough.

Reply:
Good catch, I've changed it to just `else`. (An error would be thrown a few lines beforehand if it were anything besides `EcalBarrel` or `EcalEndcap`.)